Wikipedia/Code/WebViewController.m

```diff
@@ -151,21 +151,35 @@ - (void)userContentController:(WKUserContentController*)userContentController di
                 }
             }
         } else if (message.body[@"imageClicked"]) {
-            NSNumber* imageWidth  = message.body[@"imageClicked"][@"width"];
-            NSNumber* imageHeight = message.body[@"imageClicked"][@"height"];
-            CGSize imageSize      = CGSizeMake(imageWidth.floatValue, imageHeight.floatValue);
+            NSDictionary* imageClicked = message.body[@"imageClicked"];
+            NSNumber* imageWidth       = imageClicked[@"data-file-width"] ? : imageClicked[@"width"];
+            NSNumber* imageHeight      = imageClicked[@"data-file-height"] ? : imageClicked[@"height"];
+            
+            CGSize imageSize = CGSizeZero;
+            if ([imageWidth respondsToSelector:@selector(floatValue)] && [imageHeight respondsToSelector:@selector(floatValue)]) {
+                imageSize = CGSizeMake(imageWidth.floatValue, imageHeight.floatValue);
+            }
+            
             if (![MWKImage isSizeLargeEnoughForGalleryInclusion:imageSize]) {
                 return;
             }
 
-            NSString* selectedImageURL = message.body[@"imageClicked"][@"url"];
-            NSCParameterAssert(selectedImageURL.length);
-            if (!selectedImageURL.length) {
+            NSString* selectedImageURLString = message.body[@"imageClicked"][@"url"];
+            NSCParameterAssert(selectedImageURLString.length);
+            if (!selectedImageURLString.length) {
                 DDLogError(@"Image clicked callback invoked with empty URL: %@", message.body[@"imageClicked"]);
                 return;
             }
+            
+            NSURLComponents* selectedImageURLComponents = [NSURLComponents componentsWithString:selectedImageURLString];
+            for (NSURLQueryItem* item in selectedImageURLComponents.queryItems) {
+                if ([item.name.lowercaseString isEqualToString:@"originalsrc"]) {
+                    selectedImageURLString = item.value;
```

Comment:
Here's the line:
https://github.com/wikimedia/wikipedia-ios/pull/727/files#diff-38568396230724f82d0bf5ec47cec4d8R180

I should have sync'ed up with you first but I fell down a bit of a rabbit hole :(